RecoParticleFlow/PFProducer/interface/PFAlgo.h

```diff
@@ -85,12 +85,7 @@ class PFAlgo {
                             const edm::ValueMap<reco::PhotonRef>& valueMapGedPhotons);
 
   void setPostHFCleaningParameters(bool postHFCleaning,
-                                   double minHFCleaningPt,
-                                   double minSignificance,
-                                   double maxSignificance,
-                                   double minSignificanceReduction,
-                                   double maxDeltaPhiPt,
-                                   double minDeltaMet);
+				   const edm::ParameterSet& pfHFCleaningParams);
```

Comment:
there is a preference to reduce dependence on ```ParameterSet``` this is going against that

Reply:
Hmm. What is the reason behind reducing the dependence on ParameterSet?

This PFHFCleaningParameters and also PFBadHcalMitigationParameters are less important, and we don't have to introduce these PSets, but I think it's a good idea to isolate Parameters for PFEGammaFilters and PFMuonAlgo, which I am hoping that EGamma and Muon POGs keep eyes on and manage (factorizing parameter sets to factorize responsibilities).

Reply:
I'm using this https://github.com/cms-sw/cmssw/pull/27010#issuecomment-497804713 as a reference.
Perhaps it does not fully apply in this case: there the tool is delivered via the event setup, while here the tool is created by the ED producer directly.

@Dr15Jones 
is it worth to reduce dependence on the ```ParameterSet``` in the helper algorithms used by ED producers/analyzers?
The factory pattern [not the case here, but somewhat related] even enforces this dependence.

Reply:
Okay. I perfer to leave the PR as it is, but let's see what @Dr15Jones would say.

Reply:
In general, the framework group encourages people to reduce dependencies on the ParameterSet class. It can make it harder to reuse code and the calls to `getParameter` always do parsing so are not super efficient.

Reply:
ok.